src/System.Data.SqlClient/tests/ManualTests/SQL/MirroringTest/ConnectionOnMirroringTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class ConnectionOnMirroringTest
+    {
+        [CheckConnStrSetupFact]
+        public static void TestMultipleConnectionToMirroredServer()
+        {
+            string connectionString = DataTestUtility.TcpConnStr;
+            int mirroringState;
+            string failoverPartnerName;
+            bool isMirroring = GetMirroringInfo(connectionString, out mirroringState, out failoverPartnerName);
+            List<SqlConnection> list = new List<SqlConnection>();
+            if (isMirroring && mirroringState == 4 && !string.IsNullOrEmpty(failoverPartnerName))
+            {
+                Stopwatch stopWatch = new Stopwatch();
+                TestWorker worker = new TestWorker(connectionString);
+                Thread childThread = new Thread(() => worker.TestMultipleConnection());
+
+                stopWatch.Start();
+                childThread.Start();
+                while (!worker.IsDone && stopWatch.ElapsedMilliseconds <= 10000);
+                stopWatch.Stop();
+
+                if (worker.IsDone)
+                {
+                    childThread.Join();
+                }
+                else
+                {
+                    //thread.Abort() is not implemented yet in CoreFx.
```

Comment:
Async task with timeout does not work in this case. Async task is basically run by child thread in background, and main thread uses CancellationTokenSource.Cancel() (https://msdn.microsoft.com/en-us/library/dd321955.aspx) to request cancelling the task when it is timed out. In CancellationTokenSource.Cancel(), main thread changes the cancellation token flag to true. And the child thread watches the flag, stops its work, and joins back to main thread. This only works when child thread is responsive and able to watch the cancellation token flag. In my case, the child thread gets stuck in opening connection procedure spinning infinitely so that it is no longer responsive or available to accept the flag.
I will use thread.Interrupt() for now since `thread.Abort()` is not implemented yet. Since the thread is interrupted, it does not spin anymore, just not release the memory it occupies, which does not affect the performance of next unit tests.